paddle/gserver/tests/test_NetworkCompare.cpp

```diff
@@ -116,7 +116,15 @@ void calcGradient(DataIn& in, DataOut& out, const std::string& configPath) {
   gradientMachine->start(trainer.getConfig(), nullptr);
   gradientMachine->forward(in.inArgs, &outArgs, PASS_TRAIN);
   for (size_t i = 0; i < in.outGrads.size(); i++) {
-    outArgs[i].grad->copyFrom(*in.outGrads[i]);
+    if (outArgs[i].grad) {
```

Comment:
If all the layers in the config have no parameters, or not set NeedGradient()，or the input layer have no gradient, the outArgs[i] will be nullptr. If `layer-needGradient()` is false, the backward will not  calculate in NeuralNetwork.cpp.

See `initNeedFlags()` in  `paddle/gserver/layers/Layer.cpp`.

Now add img_conv_layer in the config.